variables.tf

```diff
@@ -14,75 +14,82 @@
  * limitations under the License.
  */
 
-variable project {
+variable "project" {
   description = "The project to deploy to, if not set the default provider project is used."
   default     = ""
 }
 
-variable region {
+variable "region" {
   description = "Region for cloud resources."
   default     = "us-central1"
 }
 
-variable network {
+variable "network" {
   description = "Name of the network to create resources in."
   default     = "default"
 }
 
-variable subnetwork {
+variable "subnetwork" {
   description = "Name of the subnetwork to create resources in."
   default     = "default"
 }
 
-variable network_project {
+variable "network_project" {
   description = "Name of the project for the network. Useful for shared VPC. Default is var.project."
   default     = ""
 }
 
-variable name {
+variable "name" {
   description = "Name for the forwarding rule and prefix for supporting resources."
 }
 
-variable backends {
+variable "backends" {
   description = "List of backends, should be a map of key-value pairs for each backend, must have the 'group' key."
-  type        = "list"
+  type        = list(object({ group = string }))
 }
 
-variable session_affinity {
+variable "session_affinity" {
   description = "The session affinity for the backends example: NONE, CLIENT_IP. Default is `NONE`."
   default     = "NONE"
 }
 
-variable ports {
+variable "ports" {
   description = "List of ports range to forward to backend services. Max is 5."
-  type        = "list"
+  type        = list(string)
 }
 
-variable http_health_check {
-  description = "Set to true if health check is type HTTP, otherwise health check is TCP."
-  default     = false
+variable "health_check" {
+  type = object({
+    type                = string
+    check_interval_sec  = number
+    healthy_threshold   = number
+    timeout_sec         = number
+    unhealthy_threshold = number
+    response            = string
+    proxy_header        = string
+    health_port         = number
+    port_name           = string
+    request_path        = string
+    host                = string
+  })
 }
 
-variable health_port {
```

Comment:
Since we are breaking the interface, we will need an upgrade guide to explain how to upgrade from the last major release. Project Factory has examples of this.

Reply:
Added the upgrade guide